tree/ntuple/v7/src/RPageSinkBuf.cxx

```diff
@@ -92,10 +94,39 @@ ROOT::Experimental::Detail::RPageSinkBuf::CommitClusterImpl(ROOT::Experimental::
       fTaskScheduler->Reset();
    }
 
+   // If we have only sealed pages in all buffered columns, commit them in a single `CommitSealedPageV()` call
+   bool singleCommitCall = std::all_of(fBufferedColumns.begin(), fBufferedColumns.end(),
+                                       [](auto &bufColumn) { return bufColumn.HasSealedPagesOnly(); });
+   if (singleCommitCall) {
+      std::vector<RSealedPageGroup> toCommit;
+      toCommit.reserve(fBufferedColumns.size());
+      for (auto &bufColumn : fBufferedColumns) {
+         const auto &sealedPages = bufColumn.GetSealedPages();
+         toCommit.emplace_back(bufColumn.GetHandle().fId, sealedPages.cbegin(), sealedPages.cend());
+      }
+      fInnerSink->CommitSealedPageV(toCommit);
+
+      for (auto &bufColumn : fBufferedColumns)
+         bufColumn.DrainBufferedPages();
+      return fInnerSink->CommitCluster(nEntries);
+   }
+
+   // Otherwise, try to do it per column
    for (auto &bufColumn : fBufferedColumns) {
-      for (auto &bufPage : bufColumn.DrainBufferedPages()) {
+      auto hasSealedPagesOnly = bufColumn.HasSealedPagesOnly();
```

Comment:
Let's make sure that we have unit tests for the three cases:
  - All pages sealed
  - All pages of a column sealed
  - Some pages sealed some not